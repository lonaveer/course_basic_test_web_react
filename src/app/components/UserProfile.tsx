import React from 'react';

interface UserProfileProps {
  username: string;
  email: string;
}

const UserProfile: React.FC<UserProfileProps> = ({ username, email }) => {
  return (
    <div>
      <h1>{username}</h1>
      <p>{email}</p>
    </div>
  );
};

export default UserProfile;