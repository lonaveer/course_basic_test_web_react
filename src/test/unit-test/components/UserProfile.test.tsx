import React from 'react';
import { render } from '@testing-library/react';
import UserProfile from '../../../app/components/UserProfile';

describe('UserProfile', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<UserProfile username="loki" email="loki@test.com" />);
    expect(asFragment()).toMatchSnapshot();
  });
});