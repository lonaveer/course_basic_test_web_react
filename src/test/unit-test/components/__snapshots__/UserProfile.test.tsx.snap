// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      loki
    </h1>
    <p>
      loki@test.com
    </p>
  </div>
</DocumentFragment>
`;
